### PR TITLE
code format

### PR DESCRIPTION
**Pull request by king6cong**

### CODE REVIEW

File: src/libcore/iter/iterator.rs [hunk 1]
```diff
@@ -2144,14 +2144,14 @@ pub trait Iterator {
...
-/// Select an element from an iterator based on the given projection
+/// Select an element from an iterator based on the given "projection"
 /// and "comparison" function.
 ///
 /// This is an idiosyncratic helper to try to factor out the
 /// commonalities of {max,min}{,_by}. In particular, this avoids
 /// having to implement optimizations several times.
 #[inline]
-fn select_fold1<I,B, FProj, FCmp>(mut it: I,
+fn select_fold1<I, B, FProj, FCmp>(mut it: I,
```
Comment (malbarbo) [inline src/libcore/iter/iterator.rs]: Unaligned parameters

Comment (king6cong) [inline src/libcore/iter/iterator.rs]: Oops, updated using rustfmt

**Comment by mrhota** [comment]
@brson ping

**Comment by nagisa** [comment]
There are some debatable changes esp wrt the where clauses. I think this is the primary reason why the whole codebase not rustfmtd yet. We're also waiting for style RFCs to conclude.

**Comment by shepmaster** [comment]
> esp wrt the where clauses

If theres just a few primary concerns (like `where` clauses) perhaps the author could remove those specific ones and get the PR merged. @nagisa do you know which specific things are objectionable?

Also, if it's just a matter of running clippy again after the RFC style is approved, then perhaps the biggest downside would just be some (acceptable?) amount of churn?

**Comment by king6cong** [comment]
@shepmaster yeah, I will update this PR later tomorrow. Night in Beijing and girlfriend urging me to sleep :)

**Comment by king6cong** [comment]
Updated

**Comment by japaric** [comment]
I didn't see the previous version but the current changes look uncontroversial. Thanks for the PR, @king6cong.

@bors r+

**Pull request merged**